src/Parsers/PRQL/ParserPRQLQuery.cpp

```diff
@@ -0,0 +1,84 @@
+#include <string>
+#include <Parsers/PRQL/ParserPRQLQuery.h>
+
+#include "Parsers/Lexer.h"
+#include "config.h"
+
+#if USE_PRQL
+#    include <prql.h>
+#endif
+
+#include <Parsers/ParserQuery.h>
+#include <Parsers/ParserSetQuery.h>
+#include <Parsers/parseQuery.h>
+#include <base/scope_guard.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int SYNTAX_ERROR;
+    extern const int SUPPORT_IS_DISABLED;
+}
+
+bool ParserPRQLQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserSetQuery set_p;
+
+    if (set_p.parse(pos, node, expected))
+        return true;
+
+#if !USE_PRQL
+    throw Exception(
+        ErrorCodes::SUPPORT_IS_DISABLED, "PRQL is not available. Rust code or PRQL itself may be disabled. Use another dialect!");
+#else
+    const auto * begin = pos->begin;
+
+    while (!pos->isEnd() && pos->type != TokenType::Semicolon)
+        ++pos;
+
+    const auto * end = pos->begin;
+
+    uint8_t * sql_query_ptr{nullptr};
+    uint64_t sql_query_size{0};
+
+    const auto res
+        = prql_to_sql(reinterpret_cast<const uint8_t *>(begin), static_cast<uint64_t>(end - begin), &sql_query_ptr, &sql_query_size);
+
+    SCOPE_EXIT({ prql_free_pointer(sql_query_ptr); });
+
+    const auto * sql_query_char_ptr = reinterpret_cast<char *>(sql_query_ptr);
+    const auto * const original_sql_query_ptr = sql_query_char_ptr;
+
+    if (res != 0)
```

Comment:
I guess worth to ensure that `sql_query_size` is at least 1 byte, since there is subtraction below.